unimod_mapper/unimod_mapper.py

```diff
@@ -688,6 +719,165 @@ def _reparseXML(self, xml_file_list=[]):
         self._data_list = self._parseXML(xml_file_list=xml_file_list)
         self._mapper = self._initialize_mapper()
 
+    def map_mods(self, mod_list):
+        """
+        Maps modifications defined in params["modification"] using unimod. Using the
+        dict format for the mods, the dict can be adjusted depending on the purpose of
+        the mapping. Moreover, it can only have the minimal amound of items, or e.g.
+        engine-specific ones. At the end the mapped vallues will be updated to the
+        original dict.
+
+        Args:
+            mod_list (list): list of mod_dicts containing all relevant info about a
+                             given modification
+
+        Returns:
+            rdict (dict): dict with mod types as keys and corresponding lists of
+                             mod dicts mapped to unimod
+
+        Examples:
+
+            mod_list = [
+                {
+                    "aa": "M",              # specify the modified amino acid as a single letter, use '*' if the amino acid is variable
+                    "type": "opt",          # specify if it is a fixed (fix) or potential (opt) modification
+                    "position": "any",      # specify the position within the protein/peptide (Prot-N-term, Prot-C-term), use 'any' if the positon is variable
+                    "name": "Oxidation",    # specify the unimod PSI-MS Name (alternative to id)
+                    "id": None,             # specify the unimod Accession (alternative to name)
+                    "composition": None,    # For user-defined mods composition needs to be given as a Hill notation
+                }
+            ]
+
+        """
+
+        rdict = {"fix": [], "opt": []}
+        for index, mod in enumerate(mod_list):
+
+            # Check if any key from the mod dict is not valid!
+            allowed_keys = ["aa", "type", "position", "name", "id", "composition",
+                            "max_num_per_peptide", "intern_dist", "required", "neutral_loss"]
+            wrong_keys = []
+            for key in mod.keys():
+                if key not in allowed_keys:
+                    wrong_keys.append(key)
+            if wrong_keys != []:
+                logger.warning(f"One or more used keys are not supported. Please "
+                             f"confirm that you are using keys from {allowed_keys} "
+                             f"within your mod_dict. Continue without {mod}!")
+                # print(mod)
+                continue
+
+            unimod = False
+            unimod_id = None
+            type = mod["type"]
```

Comment:
Shall we check here if the input is one of the allowed ones?

Reply:
I included the check and logged it. 

However, there is no follow up action yet... should it be exited due to wrong input? otherwise it will be kicked out/ignored by the engine wrappers as they are hard coded to use only "fix" and "opt"...